### PR TITLE
Comparing and Sorting: Additions and reordering

### PR DESCRIPTION
**Pull request by encukou**
I started doing some light edits, but then it grew into rewriting a lot more that I originally wanted. Sorry for that; next time I'll try to do more interaction on the pull request.

### CODE REVIEW

File: source/comparisons.rst [hunk 1]
```diff
@@ -129,13 +113,120 @@ Final implementation might look like this::
...
+    >>> sorted(actors, cmp=cmp_last_name)
+    ['John Cleese', 'Terry Gilliam', 'Eric Idle', 'Terry Jones', 'Michael Palin']
+
+This function is called many times – O(*n* log *n*) – during the comparison.
+
+As an alternative to *cmp*, sorting functions can take a keyword-only ``key``
+parameter, a function that returs the key under which to sort::
+
+    >>> def keyfunction(item):
+    ...     """Key for comparison that returns a reversed string"""
```
Comment (frenzymadness) [inline source/comparisons.rst]: Forgotten comment change

File: source/comparisons.rst [hunk 2]
```diff
@@ -129,13 +113,120 @@ Final implementation might look like this::
...
+one thinks of sorting by some aspect of an object (such as last name),
+rather than by comparing individual objects.
+The main disdvantage is that the old *cmp* style is commonly used in
+C-language APIs, so external libraries are likely to provide similar functions.
+
+In Python 3, the ``cmp`` parameter was removed, and only ``key`` (or no
+argument at all) can be used.
+
+There is no fixer for this change.
+However, discovering it is straightforward: the call ing ``sort`` with the
```
Comment (frenzymadness) [inline source/comparisons.rst]: Extra space in "call ing"

**Review by frenzymadness — CHANGES_REQUESTED** [review]
Just two small changes are needed.

Thank you for your work. Unfortunately, almost all my content is gone :( so I will try to do a better job next time.

File: source/comparisons.rst (hunk 1 above)
Comment (encukou) [reply source/comparisons.rst]: Fixed & rebased

File: source/comparisons.rst (hunk 2 above)
Comment (encukou) [reply source/comparisons.rst]: Fixed & rebased

**Review by frenzymadness — APPROVED** [review]

**Pull request merged**